variables.tf

```diff
@@ -1,9 +1,212 @@
+}
+variable "reloader_is_openshift" {
+  description = "Enable OpenShift DeploymentConfigs"
+  type        = bool
+  default     = true
+}
+variable "reloader_is_argo_rollouts" {
+  description = "Enable Argo Rollouts"
+  type        = bool
+  default     = false
+}
+variable "reloader_reload_on_create" {
+  description = "Enable reload on create events"
+  type        = bool
+  default     = true
+
+}
+variable "reloader_sync_after_restart" {
+  description = "Enable sync after Reloader restarts for Add events, works only when reloadOnCreate is true"
+  type        = bool
+  default     = true
+
+}
+variable "reloader_pod_monitor_metrics" {
+  description = "Enable to scrape Reloader's Prometheus metrics"
+  type        = bool
+  default     = false
+}
+
+variable "reloader_log_format" {
+  description = "The log format to use for reloader. Possible values are `json` or `text`. Default value is `json`"
+  type        = string
+  default     = "text"
+  validation {
+    condition     = contains(["json", "text"], var.reloader_log_format)
+    error_message = "The specified reloader_log_format is not a valid selection! Valid values are `json` or `text`"
+  }
+}
+variable "reloader_custom_values" {
+  description = "String containing custom values to be used for reloader helm chart. See https://github.com/stakater/Reloader/blob/master/deployments/kubernetes/chart/reloader/values.yaml"
+  type        = string
+  default     = null
+}
+
+# image registry and digest
+
+variable "eso_registry_namespace_image" {
+  type        = string
+  description = "The External Secrets Operator image reference in the format of `[registry-url]/[namespace]/[image]`."
+  default     = "ghcr.io/external-secrets/external-secrets"
+}
+
+variable "eso_image_digest" {
+  type        = string
+  description = "The image digest in the format sha256:xxxxx... for ESO image to deploy. If not provided, a default value will be used."
+  default     = "sha256:d38834043de0a4e4feeac8a08d0bc96b71ddd7fe1d4c8583ee3751badeaeb01d" # datasource: ghcr.io/external-secrets/external-secrets
+  validation {
+    condition     = var.eso_image_digest == null || can(regex("^sha256:", var.eso_image_digest))
+    error_message = "If provided, the value of eso_image_digest must start with 'sha256:'."
+  }
+}
+
+variable "reloader_registry_namespace_image" {
+  type        = string
+  description = "The Reloader image reference in the format of `[registry-url]/[namespace]/[image]`."
+  default     = "ghcr.io/stakater/reloader"
+}
+
+variable "reloader_image_digest" {
+  type        = string
+  description = "The image digest in the format sha256:xxxxx... the reloader image to deploy. If not provided, a default value will be used."
+  default     = "sha256:80a557100c6835c7e3c9842194250c9c4ca78f43200bc3a93a32e5b105ad11bb" # datasource: ghcr.io/stakater/reloader
+  validation {
+    condition     = var.reloader_image_digest == null || can(regex("^sha256:", var.reloader_image_digest))
```

Comment:
this value will never be null, so validation should not include allowing null. Suggest to set `nullable = false`